Persist torch.assert in aten graph

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100101

This PR introduces a new operator called aten._assert_async.msg, which allows passing a tensor value and assertion message as inputs. As part of TorchDynamo, we're replacing the use of torch._assert with this new operator so that make_fx also knows how to handle assertions. This is subset of https://github.com/pytorch/pytorch/pull/98878, refer there for historic reviews. 

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

tugsbayasgalan committed:
Persist torch.assert in aten graph

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Persist torch.assert in aten graph"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Persist torch.assert in aten graph"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

jansel commented on torch/_inductor/graph.py:
```diff
@@ -356,6 +356,11 @@ def call_function(self, target, args, kwargs):
 
+        # TODO (tmanlaibaatar) Skip now because we don't have
+        # inductor lowering for this op.
+        if target == torch.ops.aten._assert_async.msg:
```
It would be cleaner to do this in `decomposition.py`

tugsbayasgalan replied:
yep! It just decomposes to nothing now

tugsbayasgalan committed:
Update on "Persist torch.assert in aten graph"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Persist torch.assert in aten graph"


This PR introduces a new operator called aten._assert_async.msg, which allows passing a tensor value and assertion message as inputs. As part of TorchDynamo, we're replacing the use of torch._assert with this new operator so that make_fx also knows how to handle assertions. This is subset of https://github.com/pytorch/pytorch/pull/98878, refer there for historic reviews. 

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge
